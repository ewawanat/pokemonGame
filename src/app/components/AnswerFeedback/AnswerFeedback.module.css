.feedbackContainer {
    margin-top: 20px;
    text-align: center;
}

.feedbackText {
    margin-bottom: 10px;
    font-size: 1.2rem;
    color: #333;
}