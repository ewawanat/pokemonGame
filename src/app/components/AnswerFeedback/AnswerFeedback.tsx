import React from 'react';
import styles from './AnswerFeedback.module.css';

type AnswerFeedbackProps = {
    userAnswer: string | null;
    correctAnswer: string | undefined;
};

const AnswerFeedback: React.FC<AnswerFeedbackProps> = ({
    userAnswer,
    correctAnswer,
}) => {
    return (
        <div className={styles.feedbackContainer}>
            <p className={styles.feedbackText}>
                {userAnswer === correctAnswer ? 'Correct!' : 'Wrong!'} It was {correctAnswer}.
            </p>
        </div>
    );
};

export default AnswerFeedback;
