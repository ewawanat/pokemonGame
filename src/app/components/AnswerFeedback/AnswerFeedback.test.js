import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { axe, toHaveNoViolations } from 'jest-axe';
import AnswerFeedback from './AnswerFeedback';

// Extend Jest to include axe accessibility checks
expect.extend(toHaveNoViolations);

describe('AnswerFeedback Component', () => {
    const defaultProps = {
        userAnswer: null,
        correctAnswer: 'A',
    };

    const renderComponent = (props = {}) => {
        return render(<AnswerFeedback {...defaultProps} {...props} />);
    };

    it('renders the correct feedback when the answer is correct', () => {
        renderComponent({
            userAnswer: 'A',
            correctAnswer: 'A',
        });

        const feedbackText = screen.getByText(/correct!/i);
        expect(feedbackText).toBeInTheDocument();
        expect(feedbackText).toHaveTextContent('Correct! It was A.');
    });

    it('renders the wrong feedback when the answer is incorrect', () => {
        renderComponent({
            userAnswer: 'B',
            correctAnswer: 'A',
        });

        const feedbackText = screen.getByText(/wrong!/i);
        expect(feedbackText).toBeInTheDocument();
        expect(feedbackText).toHaveTextContent('Wrong! It was A.');
    });

    it('renders nothing when no answer is provided', () => {
        renderComponent({
            userAnswer: null,
            correctAnswer: 'A',
        });

        const feedbackText = screen.queryByText(/correct!/i);
        expect(feedbackText).not.toBeInTheDocument();
    });

    it('matches the snapshot when the answer is correct', () => {
        const { asFragment } = renderComponent({
            userAnswer: 'A',
            correctAnswer: 'A',
        });
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches the snapshot when the answer is incorrect', () => {
        const { asFragment } = renderComponent({
            userAnswer: 'B',
            correctAnswer: 'A',
        });
        expect(asFragment()).toMatchSnapshot();
    });

    it('has no accessibility violations', async () => {
        const { container } = renderComponent({
            userAnswer: 'A',
            correctAnswer: 'A',
        });
        const results = await axe(container);
        expect(results).toHaveNoViolations();
    });

    it('has no accessibility violations when the answer is wrong', async () => {
        const { container } = renderComponent({
            userAnswer: 'B',
            correctAnswer: 'A',
        });
        const results = await axe(container);
        expect(results).toHaveNoViolations();
    });
});
