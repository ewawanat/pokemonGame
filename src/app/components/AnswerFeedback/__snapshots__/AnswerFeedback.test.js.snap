// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerFeedback Component matches the snapshot when the answer is correct 1`] = `
<DocumentFragment>
  <div
    class="feedbackContainer"
  >
    <p
      class="feedbackText"
    >
      Correct! It was A.
    </p>
  </div>
</DocumentFragment>
`;

exports[`AnswerFeedback Component matches the snapshot when the answer is incorrect 1`] = `
<DocumentFragment>
  <div
    class="feedbackContainer"
  >
    <p
      class="feedbackText"
    >
      Wrong! It was A.
    </p>
  </div>
</DocumentFragment>
`;
